community/kernel/src/main/java/org/neo4j/kernel/api/schema/index/IndexDescriptor.java

```diff
@@ -36,15 +36,20 @@
 /**
  * Internal representation of a graph index, including the schema unit it targets (eg. label-property combination)
  * and the type of index. UNIQUE indexes are used to back uniqueness constraints.
- *
- * Please note that this class has dependencies in other modules. See {@see org.neo4j.causalclustering.catchup.storecopy.IndexDescriptorSerializer}
  */
-public class IndexDescriptor implements LabelSchemaSupplier
+public abstract class IndexDescriptor implements SchemaDescriptorSupplier
 {
+
+    public String metadata()
+    {
+        return "";
+    }
+
     public enum Type
     {
         GENERAL,
-        UNIQUE
+        UNIQUE,
+        NON_SCHEMA
```

Comment:
this should be renamed together with all related things to FULL_TEXT

Reply:
This isn't called full_text because it is not necessarily fulltext specific. I mean, it is right now, but there is nothing stating that is has to be, it could be any index that isn't a schema index. There is also no mention of fulltext anywhere in the kernel code as a whole, all of the fulltext aware code lives in the fulltext module, to me it feels like a nice thing that it doesn't bleed into the kernel code.